contrib/hbcurl/core.c

```diff
@@ -1034,6 +1051,50 @@ HB_FUNC( CURL_EASY_SETOPT )
                }
             }
             break;
+            case HB_CURLOPT_HTTPPOST_FORM:
+            {
+               PHB_ITEM pArray = hb_param( 3, HB_IT_ARRAY );
+
+               if( pArray )
+               {
+                  HB_SIZE ulPos;
+                  HB_SIZE ulArrayLen = hb_arrayLen( pArray );
+
+                  for( ulPos = 0; ulPos < ulArrayLen; ++ulPos )
+                  {
+                     PHB_ITEM pSubArray = hb_arrayGetItemPtr( pArray, ulPos + 1 );
+
+                     int type = hb_arrayGetNI( pSubArray, 1 );
+
+                     switch( type )
+                     {
+                        case 1:
+
+                           curl_formadd( &hb_curl->pHTTPPOST_First,
+                                         &hb_curl->pHTTPPOST_Last,
+                                         CURLFORM_COPYNAME, hb_arrayGetCPtr( pSubArray, 2 ),
+                                         CURLFORM_NAMELENGTH, hb_arrayGetCLen( pSubArray, 2 ),
+                                         CURLFORM_COPYCONTENTS, hb_arrayGetCPtr( pSubArray, 3 ),
+                                         CURLFORM_CONTENTSLENGTH, hb_arrayGetCLen( pSubArray, 3 ),
+                                         CURLFORM_END );
+
+                           break;
+                        case 2:
```

Comment:
This should work as file upload from memory, with all parameters set:
```
    CURLFORM_COPYNAME, "file",
    CURLFORM_BUFFER, "unnamed.png",
    CURLFORM_BUFFERPTR, memblock,
    CURLFORM_BUFFERLENGTH, lsize,
    CURLFORM_CONTENTTYPE, "image/png",
```

To replace `CURLFORM_FILE` with something that doesn't require `FILE` structures to be passed around and can plug into the Harbour file I/O instead, there is `CURLFORM_STREAM`.

Still to be solved to somehow differentiate these parts with a numeric. But that numeric should be added to `hbcurl.ch` with `HB_` prefix, f.e. `HB_CURLFORMTYPE_*`.